presto-druid/src/main/java/com/facebook/presto/druid/authentication/DruidAuthenticationModule.java

```diff
@@ -26,49 +27,68 @@
         extends AbstractConfigurationAwareModule
 {
     @Override
     protected void setup(Binder binder)
     {
+        DruidConfig druidConfig = buildConfigObject(DruidConfig.class);
+
         bindAuthenticationModule(
                 config -> config.getDruidAuthenticationType() == NONE,
-                noneAuthenticationModule());
+                noneAuthenticationModule(druidConfig));
 
         bindAuthenticationModule(
                 config -> config.getDruidAuthenticationType() == BASIC,
-                basicAuthenticationModule());
+                basicAuthenticationModule(druidConfig));
 
         bindAuthenticationModule(
                 config -> config.getDruidAuthenticationType() == KERBEROS,
-                kerberosbAuthenticationModule());
+                kerberosbAuthenticationModule(druidConfig));
     }
 
     private void bindAuthenticationModule(Predicate<DruidConfig> predicate, Module module)
     {
         install(installModuleIf(DruidConfig.class, predicate, module));
     }
 
-    private static Module noneAuthenticationModule()
+    private static Module noneAuthenticationModule(DruidConfig druidConfig)
     {
-        return binder -> httpClientBinder(binder).bindHttpClient("druid-client", ForDruidClient.class);
+        return binder -> httpClientBinder(binder).bindHttpClient("druid-client", ForDruidClient.class)
+                .withConfigDefaults(config -> applyTlsConfig(config, druidConfig));
     }
 
-    private static Module basicAuthenticationModule()
+    private static Module basicAuthenticationModule(DruidConfig druidConfig)
     {
         return binder -> httpClientBinder(binder).bindHttpClient("druid-client", ForDruidClient.class)
                 .withConfigDefaults(
-                        config -> config.setAuthenticationEnabled(false) //disable Kerberos auth
+                        config -> {
+                            config.setAuthenticationEnabled(false); //disable Kerberos auth
+                            applyTlsConfig(config, druidConfig);
+                        }
                 ).withFilter(
                         DruidBasicAuthHttpRequestFilter.class);
     }
 
-    private static Module kerberosbAuthenticationModule()
+    private static Module kerberosbAuthenticationModule(DruidConfig druidConfig)
     {
         return binder -> httpClientBinder(binder).bindHttpClient("druid-client", ForDruidClient.class)
                 .withConfigDefaults(
-                        config -> config.setAuthenticationEnabled(true));
+                        config -> {
+                            config.setAuthenticationEnabled(true);
+                            applyTlsConfig(config, druidConfig);
+                        });
+    }
+
+    private static void applyTlsConfig(HttpClientConfig config, DruidConfig druidConfig)
+    {
+        if (druidConfig.isTlsEnabled()) {
+            requireNonNull(druidConfig.getTrustStorePath(), "TrustStorePath is null");
```

Comment:
Let's give the user-facing config name in the error message
```suggestion
            requireNonNull(druidConfig.getTrustStorePath(), "druid.tls.truststore-path is null");
```

Reply:
Agree. Updated accordingly.